File: catkin_ws/src/my_robot_tutorials/src/hw_status_subscriber.cpp
#include <ros/ros.h>
#include <my_robot_msgs/HardwareStatus.h>


void cb_recieve_status(const my_robot_msgs::HardwareStatus &data)
{
    ROS_INFO("Temperature : %i, Up/Down: %s, Msg: %s",
            (int)data.temperature,
            data.are_motors_up ? "True" : "False",
            data.debug_msg.c_str()
    );
}


int main(int argc, char* argv[])
{
    ros::init(argc, argv, "status_sub");
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("/my_robot/hardware_status",
                                        10,
                                        cb_recieve_status);

    ros::spin();
}

File: catkin_ws/src/my_robot_tutorials/scripts/hw_status_subscriber.py
#! /usr/bin/env python

import rospy
from my_robot_msgs.msg import HardwareStatus

def cb_recieve_status(data:HardwareStatus):
    rospy.loginfo(f"Motors Up: {data.are_motors_up},"
                  f"Motor Temp: {data.temperature},"
                  f"Message: {data.debug_msg}")

if __name__ == "__main__":
    rospy.init_node("status_sub")

    status_sub = rospy.Subscriber("/my_robot/hardware_status",
                                  HardwareStatus,
                                  cb_recieve_status)
    
    rospy.spin()